### DISCUSSION TITLE
Compatible with libc++?

### DISCUSSION THREAD

**Discussion by Burnrate** [foonathan/memory#116] [Q&A] [answered]
I've been trying to build a static library for aarch64 using libc++.  I've been getting lots of undefined reference errors related to std functions.  I've been working on my build process for a few days but can't seem to find anything wrong with it.  The build process works for making other static libraries for aarch64 using libc++.

Any reason using libc++ might cause problems?

**Comment by foonathan** [comment, marked as answer]
No idea. The library is tested on libc++ on MacOS and it works there.

**Reply by Burnrate** [reply]
Edit 2: We found when we cloned the repo to build it we were copying in some incompatible flags with the set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ...  .  It still doesn't fit in our build process because we haven't cross compiled from x64 to aarch64 but we can build it and use it on aarch64.

...............
Edit: Still haven't figured this out, my previous post below resulted in a weird unusable success for a while.

............
In cmake we had to set CXX_EXTENSTIONS OFF and add fuse-ld=lld to the link options for this specific cmake file.

We had already set it up to use the correct compiler and our setup worked for other libraries but 🤷
